### PR TITLE
Introduce kludge to make "isEnabledForCluster" work again for kube object details PART 6

### LINKED ISSUE #4632: KubeObjectDetailRegistry

### PR DESCRIPTION
**Pull request by jansav**
Part 6 of #5800 

In this PR, we replace `KubeObjectDetailsRegistry` with new alternative which works properly with `isEnabledForCluster`.

Signed-off-by: Janne Savolainen <janne.savolainen@live.fi>

Closes #4632

### CODE REVIEW

File: src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-matches-to-kind-and-api-version.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { KubeObject } from "../../../../common/k8s-api/kube-object";
+
+export const kubeObjectMatchesToKindAndApiVersion =
+  (kind: string, apiVersions: string[]) =>
+  <TKubeObject extends KubeObject>(item?: TKubeObject) =>
```
Comment (Nokel81) [inline src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-matches-to-kind-and-api-version.ts]: nit: I think you meant to have `item: TKubeObject | undefined`

File: src/renderer/components/kube-object-details/kube-object-detail-items/implementations/custom-resource-definitions-detail-item.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import currentKubeObjectInDetailsInjectable from "../../current-kube-object-in-details.injectable";
+
+const customResourceDefinitionsDetailItemInjectable = getInjectable({
+  id: "custom-resource-definitions-detail-item",
+
+  instantiate: (di) => {
+    const kubeObject = di.inject(currentKubeObjectInDetailsInjectable);
+
+    return {
+      Component: CRDDetails,
```
Comment (Nokel81) [inline src/renderer/components/kube-object-details/kube-object-detail-items/implementations/custom-resource-definitions-detail-item.injectable.ts:19]: nit: spacing.

File: src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import { isPod } from "./pod-detail-item.injectable";
+import { isReplicaSet } from "./replica-set-detail-item.injectable";
+import { isRole } from "./role-detail-item.injectable";
+import { isRoleBinding } from "./role-binding-detail-item.injectable";
+import { isService } from "./service-detail-item.injectable";
+import { isServiceAccount } from "./service-account-detail-item.injectable";
+import { isStatefulSet } from "./stateful-set-detail-item.injectable";
+import { isStorageClass } from "./storage-class-detail-item.injectable";
+
+const kubeEventForClusterRoleDetailItemInjectable = getInjectable({
```
Comment (Nokel81) [inline src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts]: I think this name is wrong, I don' think this has anything to do with `ClusterRole`'s

File: src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-detail-item-injection-token.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import type { KubeObjectDetailsProps } from "../kube-object-details";
+import type React from "react";
+
+export interface KubeObjectDetailItem {
+  Component: React.ElementType<KubeObjectDetailsProps<any>>;
```
Comment (Nokel81) [inline src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-detail-item-injection-token.ts:11]: Same comment as from a previous PR about `KubeObject`

**Review by Nokel81 — APPROVED** [review]

File: src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts (hunk 3 above)
Comment (jansav) [reply src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts]: Yep, it is. Good catch.

File: src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-matches-to-kind-and-api-version.ts (hunk 1 above)
Comment (jansav) [reply src/renderer/components/kube-object-details/kube-object-detail-items/kube-object-matches-to-kind-and-api-version.ts]: Thanks.

File: src/renderer/components/kube-object-details/kube-object-detail-items/implementations/custom-resource-definitions-detail-item.injectable.ts (hunk 2 above)
Comment (jansav) [reply src/renderer/components/kube-object-details/kube-object-detail-items/implementations/custom-resource-definitions-detail-item.injectable.ts:19]: Fixed

File: src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts (hunk 3 above)
Comment (jansav) [reply src/renderer/components/kube-object-details/kube-object-detail-items/implementations/kube-event-detail-item.injectable.ts]: Fixed

**Pull request merged**